### ISSUE TITLE
compiler include directory flag incorrect on windows

### ISSUE THREAD

**Issue by boxerab** [mesonbuild/meson#6794]
**Describe the bug**
build.ninja has `-I` as include directory directive - this should be translated to `/I`

**To Reproduce**
Run `meson build` on windows and then look at `build.ninja` file

**Expected behavior**
`-I` should in fact be `/I`

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer)?
* what operating system (e.g. MacOS Catalina, Windows 10, CentOS 8.0, Ubuntu 18.04, etc.)
* what Python version are you using e.g. 3.8.0
* what `meson --version`
* what `ninja --version` if it's a Ninja build

**Comment by xclaesse** [comment]
@nirbheek you're the one with MSVC experience.

I've been talking with @boxerab and it seems the issue is when `-I` flags comes from pkg-config.

**Comment by boxerab** [comment]
@xclaesse the include flag for subprojects are also  `-I`, but perhaps this is handled correctly.

**Comment by dcbaker** [comment]
This sounds like some or all dependencies aren't running their arguments through the `Compiler`'s `to_native` method (I might have the name of the method wrong, but it's something like that)

**Comment by xclaesse** [comment]
It does not seems visualstudio compiler convert them at all: https://github.com/mesonbuild/meson/blob/master/mesonbuild/compilers/mixins/visualstudio.py#L216. I have no idea how that's supposed to work... git blame says @dcbaker and @mensinda  worked on that code.

**Comment by dcbaker** [comment]
I just moved that code around, I've never done real work on it.

**Comment by boxerab** [comment]
@dcbaker so, `nasm` has a different `Compiler` ?  Because `nasm` uses unix-style `-I` to specify include directories, so these should be left along.

**Comment by dcbaker** [comment]
Yes, NASM should be represented with a different `Compiler` (I haven't looked at NASM to be sure). Generally meson internally uses Posix/Unix/Gnu style flags and then translates them to the native flags in the backend.

**Comment by bonzini** [comment]
Fixed, see `VisualStudioLikeCompiler.include_arg_to_native` for example.
